### PR TITLE
Add reproducable theoretical max TPS test

### LINKED ISSUE #170: Formalize and publish performance evaluation methodology
This issue tracks completing and documenting a performance evaluation framework to measure maximum theoretical throughput of the Stellar network. A couple of items here: 
* Complete https://github.com/stellar/supercluster/pull/164 which adds support for Soroban+classic load
* Open source and document various topologies to use for TPS tests
* Better document the existing (and new) TPS test, with steps and configurations needed to reproduce
* Include hardware/cloud recommendations

### PR DESCRIPTION
**Pull request by bboston7**
Closes #170

This change adds documentation on how to run a theoretical max TPS test and publishes our first set of results. It also adds a script to enable anyone to spin up a supercluster-compatable EKS instance so that they may reproduce our results.

### CODE REVIEW

File: doc/theoretical-max-tps.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+Finally, don't forget to shut down your EKS cluster.
+
+## Results
+
+This table contains the theoretical max TPS stellar-core achieved, ordered by
+stellar-core release.
+
+| Core Version | Core Image | EC2 Instance Type | Number of Instances | Max TPS |
+|--------------|------------|-------------------|---------------------|---------|
+| 21.1.0 | `stellar/unsafe-stellar-core:21.0.1-1917.52a449ff3.focal-testing-asan-disabled-perftests` | m5d.4xlarge | 10 | 1137 |
```
Comment (bboston7) [inline doc/theoretical-max-tps.md:31]: We had another release while these tests were running (21.2.0), so we should run the tests on that release too. Someone other than me could do that as a way to test the instructions. Otherwise I'm happy to do that. Regardless, those results can go up in a separate PR if we want to get this merged sooner.

**Review by marta-lokhova — APPROVED** [review]
Looks great!

File: doc/eks.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+### Set up AWS CLI tools
+
+Before running the script, please follow
+[Amazon's guide for setting up tools to manage EKS clusters](https://docs.aws.amazon.com/eks/latest/userguide/setting-up.html).
+
+### AWS Quotas
+
+In order to run the theoretical max TPS test with the recommended parameters,
+you'll need a quota of at least 160 vCPUs for on-demand M instance types in the
```
[the remark this answers is not in the document]
Comment (bboston7) [reply doc/eks.md:24]: The instance type we recommend (`m5d.4xlarge`) has 16 vCPUs, and since we need to spin up 10 of them the AWS account needs to have a quota of at least 160. It's true that stellar-core doesn't use all of those vCPUs, but AWS bundles a bunch of other stuff we want with the larger instance (higher bandwidth, more IOPS, etc). If you're curious, there's a breakdown of all the stuff that gets bundled with each instance type here: https://docs.aws.amazon.com/ec2/latest/instancetypes/gp.html

[the remark this answers is not in the document]
Comment (marta-lokhova) [reply doc/eks.md:24]: Makes sense, thanks!

File: doc/theoretical-max-tps.md (hunk 1 above)
Comment (bboston7) [reply doc/theoretical-max-tps.md:31]: Marking resolved so github will let me merge this. We will add 21.2.0 results shortly in a separate PR
